extensions/pkg/predicate/predicate.go

```diff
@@ -306,3 +305,31 @@ func ClusterShootKubernetesVersionForCSIMigrationAtLeast(decoder runtime.Decoder
 		},
 	}
 }
+
+// ShootIsUnassigned is a predicate that returns true if a shoot is not assigned to a seed.
```

Comment:
I started to move out the affected predicates to `/pkg/predicate` but left unaffected predicates in `/extensions/pkg/predicate` because I see the necessity rewrite some predicates which we should do in a separate PR. I opened https://github.com/gardener/gardener/issues/4487 as a follow-up item.